warehouse/query-core/src/test/java/datawave/query/UniqueTest.java

```diff
@@ -351,4 +354,34 @@ public void testUniquenessWithModelAliases() throws Exception {
         String queryString = "UUID:/^[CS].*/ AND #UNIQUE(BOTH_NULL)";
         runTestQueryWithUniqueness(expected, queryString, startDate, endDate, extraParameters);
     }
+
+    @Test
+    public void testMostRecentUniqueness() throws Exception {
+        Map<String,String> extraParameters = new HashMap<>();
+        extraParameters.put("include.grouping.context", "true");
+        extraParameters.put(QueryParameters.MOST_RECENT_UNIQUE, "true");
+
+        Date startDate = format.parse("20091231");
+        Date endDate = format.parse("20150101");
+
+        String queryString = "UUID =~ '^[CS].*'";
+
+        Set<Set<String>> expected = new HashSet<>();
+        expected.add(Sets.newHashSet(WiseGuysIngest.caponeUID));
+        extraParameters.put("unique.fields", "DEATH_DATE,$MAGIC");
+        runTestQueryWithUniqueness(expected, queryString, startDate, endDate, extraParameters);
+
+        expected.add(Sets.newHashSet(WiseGuysIngest.sopranoUID));
+        expected.add(Sets.newHashSet(WiseGuysIngest.corleoneUID));
+        expected.add(Sets.newHashSet(WiseGuysIngest.caponeUID));
+        extraParameters.put("unique.fields", "$DEATH_DATE,BIRTH_DATE");
+        runTestQueryWithUniqueness(expected, queryString, startDate, endDate, extraParameters);
+
+        expected.add(Sets.newHashSet(WiseGuysIngest.sopranoUID));
+        expected.add(Sets.newHashSet(WiseGuysIngest.corleoneUID));
+        expected.add(Sets.newHashSet(WiseGuysIngest.caponeUID));
+        extraParameters.put("unique.fields", "death_date,birth_date");
+        runTestQueryWithUniqueness(expected, queryString, startDate, endDate, extraParameters);
+    }
+
```

Comment:
I might have missed it, but do we have a test to make sure this also works with the #UNIQUE variant, just to make sure it doesn't break in future?

Reply:
Maybe some negative cases, too

Reply:
Yes, we do have #UNIQUE tests in this very class.

Reply:
What do you mean by negative cases?

Reply:
No, I meant that in addition to a test adding just unique.fields that we should have an additional test with most recent set to true using #UNIQUE or f:unique in the query.  I saw that there were already standalone tests of these... thanks....  to further elaborate, I would expect

Q1:

- extraParameters.put(QueryParameters.MOST_RECENT_UNIQUE, "true");
- extraParameters.put("unique.fields", "DEATH_DATE,$MAGIC");
- String queryString = "UUID =~ '^[CS].*'";

To return the same results as

Q2:
- extraParameters.put(QueryParameters.MOST_RECENT_UNIQUE, "true");
- String queryString = "UUID =~ '^[CS].*' && f:unique(death_date,magic)";

but, for some reason that does not appear to be happening in my local testing.

By negative testing, I mean supplying invalid or contradictory input to confirm desired behavior / error handling.

Reply:
You are absolutely correct, the most recent flag is getting cleared in the second query.  Fixing